cpp/src/ffi_go/ir/deserializer.cpp

```diff
@@ -231,10 +244,71 @@ auto deserialize_wildcard_match(
         log_event->m_log_message.m_data = deserializer->m_log_event.m_log_message.data();
         log_event->m_log_message.m_size = deserializer->m_log_event.m_log_message.size();
         log_event->m_timestamp = deserializer->m_timestamp;
+        log_event->m_utc_offset = static_cast<epoch_time_ms_t>(deserializer->m_utc_offset.count());
         *matching_query = matching_query_idx;
         return static_cast<int>(IRErrorCode::IRErrorCode_Success);
     }
 }
+
+auto try_deserialize_utc_offset_changes(
+        BufferReader& ir_buf,
+        encoded_tag_t& tag,
+        Deserializer* deserializer
+) -> IRErrorCode {
+    if (UtcOffsetChange != tag) {
+        return IRErrorCode::IRErrorCode_Success;
+    }
+
+    UtcOffset utc_offset{0};
+    while (UtcOffsetChange == tag) {
```

Comment:
Lets just simplify this by taking out the while loop and drop the `try` part. If we find multiple offset change packets in a row it will get handled by the next log event while loop anyway, so lets just keep this function as simple as possible.